test_post_request.py

```diff
@@ -17,13 +14,13 @@ def do_post_request():
     #                                              y=["leftamygdala"])
 
     # example request 2
-    algorithm_input_data = AlgorithmInputDataDTO(
-        pathology="dementia",
-        datasets=["demo_data"],
-        filters=None,
-        x=["lefthippocampus", "righthippocampus"],
-        y=["alzheimerbroadcategory_bin"],
-    )
+    algorithm_input_data = {
+        "pathology": "dementia",
+        "datasets": ["demo_data"],
+        "filter": None,
+        "x": ["lefthippocampus", "righthippocampus"],
+        "y": ["alzheimerbroadcategory_bin"],
+    }
 
     algorithm_request = AlgorithmRequestDTO(
         inputdata=algorithm_input_data, parameters=None
```

Comment:
Actually, 
1. mipengine/controller/api/AlorithmRequestDto::input_data is very different compared to 
2. mipengine/controller/algorithms_specifications.py::InputDataSpecification

the first one holds info about pathologies,datasets,filters, input variables and the second one labels,descriptions,types,statistical types, etc. but both are called input data which makes it very difficult to understand what is the purpose and need for each one of them